src/controllers/cgnOperatorSearchController.ts

```diff
@@ -106,4 +100,56 @@ export default class CgnOperatorSearchController {
           )
       )
     );
+
+  /**
+   * Get a discount bucket code by discount identifier.
+   */
+  public readonly getDiscountBucketCode = (
+    req: express.Request
+  ): Promise<
+    | IResponseErrorValidation
+    | IResponseErrorInternal
+    | IResponseErrorNotFound
+    | IResponseErrorForbiddenNotAuthorized
+    | IResponseSuccessJson<DiscountBucketCode>
+  > =>
+    withUserFromRequest(req, async user => {
+      const eligibleUserOrErrorResult = await this.eligibleUserOrError(user);
+      if (eligibleUserOrErrorResult.isLeft()) {
+        return eligibleUserOrErrorResult.value;
+      }
+
+      return withValidatedOrValidationError(
+        NonEmptyString.decode(req.params.discountId),
+        discountId =>
+          this.cgnOperatorSearchService.getDiscountBucketCode(discountId)
+      );
+    });
+
+  private readonly isCgnStatusResponseSuccess = (
+    res: IResponse<unknown>
+  ): res is IResponseSuccessJson<Card> => res.kind === "IResponseSuccessJson";
+
+  private readonly eligibleUserOrError = async (
+    user: User
+  ): Promise<
+    Either<IResponseErrorInternal | IResponseErrorForbiddenNotAuthorized, void>
+  > => {
+    const cgnStatusResponse = await this.cgnService.getCgnStatus(user);
+
+    return fromPredicate(this.isCgnStatusResponseSuccess, () =>
+      ResponseErrorInternal("Cannot retrieve cgn card status")
+    )(cgnStatusResponse)
+      .mapLeft<IResponseErrorInternal | IResponseErrorForbiddenNotAuthorized>(
+        identity
+      )
+      .map(res => res.value.status)
+      .chain(
+        fromPredicate(
+          status => status === "ACTIVATED",
+          () => ResponseErrorForbiddenNotAuthorized
+        )
+      )
+      .map(() => void 0);
```

Comment:
```suggestion
      .map(constVoid);
```

Reply:
not needed anymore, look at the current version